hack/precommit-hook.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+TOP_LEVEL=$(git rev-parse --show-toplevel)
+
+if [[ $TOP_LEVEL == "" ]]; then
+  echo "Error: this script must be called from a git repository"
+  exit 1
+fi
+
+if [[ $1 == "setup" ]]; then
+	ln -fs  $TOP_LEVEL/hack/precommit-hook.sh $TOP_LEVEL/.git/hooks/pre-commit
```

Comment:
I would prefer to not overwrite existing hooks by removing the `-f`. WDYT?

Reply:
i didn't find any pre-existing commit hooks on github, so that should b ok. done.